### ISSUE TITLE
format wandb run name in pytorch checkpoint

### ISSUE THREAD

**Issue by tobiascz** [Lightning-AI/pytorch-lightning#19317]
### Description & Motivation

It would be great to format the wandb run name: 
`wandb.run.name`

into the checkpoint filename.

epoch: 10
loss: 0.5423

filename: "epoch{epoch}-loss{val_loss:.2f}" --> "epoch10-loss0.54"

filename: "{wandb.run.name}-{epoch}-{val_loss:.2f}" --> "swept_sky_54-epoch10-loss0.54"

**Comment by awaelchli** [comment]
@tobiascz What if you include the run name directly in the filename from the beginning?

```py
logger = WandbLogger(...)
checkpoint = ModelCheckpoint(..., filename=f"{logger.experiment.name}-xyz.ckpt")
```

**Comment by tobiascz** [comment]
I am using Pytorch CLI:

`  callbacks:
    - class_path: lightning.pytorch.callbacks.ModelCheckpoint
      init_args:
        dirpath: "/home/Develop/project/checkpoints"
        monitor: "val_loss"
        save_top_k: 1
        mode: "min"
        filename: "{epoch}-{val_loss:.2f}-{wandb.run.name}"
    #  wandb.run.name not accessible`
    
this is my main:

    `def cli_main():
    cli = LightningCLI(model_class=MyModel, datamodule_class=DataModuleHTC, run=False, save_config_kwargs={"overwrite": True})
    cli.trainer.fit(cli.model, cli.datamodule)


if __name__ == "__main__":
    cli_lightning_logo()
    cli_main()`
    
    
here wandb has not been initilaized yet. I think the wandb is known once the first logging has been done.

**Comment by awaelchli** [comment]
Oh ok, this is interesting. I don't actually know how to do this or whether it is possible to do elegantly. Right now I think you would need to create the callback programmatically and call into wandb to get the run name in the main script.
